pyccel/codegen/printing/ccode.py

```diff
@@ -1647,7 +1647,10 @@ def function_signature(self, expr, print_arg_names = True):
 
         if n_results > 1 or isinstance(expr.results.var.class_type, InhomogeneousTupleType):
             ret_type = self.get_c_type(PythonNativeInt())
-            arg_vars.extend(result_vars)
+            if expr.arguments and expr.arguments[0].bound_argument:
+                arg_vars = arg_vars[:1] + result_vars + arg_vars[1:]
```

Comment:
Would you mind adding a comment here to remind the developer that `arg_vars[0]` is the class object `self` in this case? In the Fortran printer you call this variable `class_arg`, which makes it quite explicit.

Reply:
Done : cf4a46570

Reply:
Thanks